### PR TITLE
Promotion translations

### PR DESCRIPTION
**Pull request by zedzior**
It is part of: https://github.com/saleor/saleor/issues/12539
Resolves https://github.com/saleor/saleor/issues/13305

I want to merge this change, because I want to be able to translate `Promotion` and `PromotionRule` `name` and `description` fields.

This PR also adds missing descriptions for translation related types and mutations.

```graphql
type PromotionTranslation implements Node {
  id: ID!
  language: LanguageDisplay!
  name: String
  description: JSONString
}

type PromotionRuleTranslation implements Node {
  id: ID!
  language: LanguageDisplay!
  name: String
  description: JSONString
}
```

# Impact

* [x] New migrations
* [x] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [ ] Privileged queries and mutations are either absent or guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migrations are either absent or optimized for zero downtime
* [ ] The changes are covered by test cases

### CODE REVIEW

File: saleor/discount/migrations/0048_promotionruletranslation.py
Comment (IKarbowiak) [inline saleor/discount/migrations/0048_promotionruletranslation.py:1]: We can merge this migration with the previous one

File: saleor/graphql/translations/types.py [hunk 1]
```diff
@@ -627,3 +627,53 @@ class Meta:
...
+class PromotionTranslation(BaseTranslationType[discount_models.PromotionTranslation]):
+    id = graphene.GlobalID(required=True)
+    name = graphene.String()
+    description = JSONString(
+        description="Translated description of the promotion." + RICH_CONTENT
+    )
+
+    class Meta:
+        model = discount_models.Promotion
+        interfaces = [graphene.relay.Node]
```
Comment (IKarbowiak) [inline saleor/graphql/translations/types.py:641]: Lack of `ADDED_IN_X` label

File: saleor/graphql/translations/types.py [hunk 2]
```diff
@@ -627,3 +627,53 @@ class Meta:
...
+
+
+class PromotionTranslatableContent(ModelObjectType[discount_models.Promotion]):
+    id = graphene.GlobalID(required=True)
+    name = graphene.String(required=True)
+    description = JSONString(description="Description of the promotion." + RICH_CONTENT)
+    translation = TranslationField(PromotionTranslation, type_name="promotion")
+
+    class Meta:
```
Comment (IKarbowiak) [inline saleor/graphql/translations/types.py:650]: Same here 👮‍♀️

File: saleor/graphql/translations/types.py [hunk 3]
```diff
@@ -627,3 +627,53 @@ class Meta:
...
+class PromotionRuleTranslation(
+    BaseTranslationType[discount_models.PromotionTranslation]
+):
+    id = graphene.GlobalID(required=True)
+    name = graphene.String()
+    description = JSONString(
+        description="Translated description of the promotion rule." + RICH_CONTENT
+    )
+
+    class Meta:
```
Comment (IKarbowiak) [inline saleor/graphql/translations/types.py:664]: And here 👮‍♀️

File: saleor/graphql/translations/types.py [hunk 4]
```diff
@@ -627,3 +627,53 @@ class Meta:
...
+
+class PromotionRuleTranslatableContent(ModelObjectType[discount_models.Promotion]):
+    id = graphene.GlobalID(required=True)
+    name = graphene.String(required=True)
+    description = JSONString(
+        description="Description of the promotion rule." + RICH_CONTENT
+    )
+    translation = TranslationField(PromotionRuleTranslation, type_name="promotion rule")
+
+    class Meta:
```
Comment (IKarbowiak) [inline saleor/graphql/translations/types.py:677]: And here 👮‍♀️

File: saleor/graphql/translations/mutations/promotion_rule_translate.py [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+class PromotionRuleTranslate(BaseTranslateMutation):
+    class Arguments:
+        id = graphene.ID(
+            required=True,
+            description="PromotionRule ID or PromotionRuleTranslatableContent ID.",
+        )
+        language_code = graphene.Argument(
+            LanguageCodeEnum, required=True, description="Translation language code."
+        )
+        input = PromotionRuleTranslationInput(required=True)
```
Comment (fowczarek) [inline saleor/graphql/translations/mutations/promotion_rule_translate.py:26]: Missing description.

File: saleor/graphql/translations/mutations/promotion_translate.py [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+class PromotionTranslate(BaseTranslateMutation):
+    class Arguments:
+        id = graphene.ID(
+            required=True,
+            description="Promotion ID or PromotionTranslatableContent ID.",
+        )
+        language_code = graphene.Argument(
+            LanguageCodeEnum, required=True, description="Translation language code."
+        )
+        input = PromotionTranslationInput(required=True)
```
Comment (fowczarek) [inline saleor/graphql/translations/mutations/promotion_translate.py:26]: Missing description.

File: saleor/graphql/translations/types.py [hunk 7]
```diff
@@ -627,3 +627,53 @@ class Meta:
     @staticmethod
     def resolve_shipping_method(root: shipping_models.ShippingMethod, _info):
         return ChannelContext(node=root, channel_slug=None)
+
+
+class PromotionTranslation(BaseTranslationType[discount_models.PromotionTranslation]):
+    id = graphene.GlobalID(required=True)
```
Comment (fowczarek) [inline saleor/graphql/translations/types.py:633]: Missing description.

File: saleor/graphql/translations/types.py [hunk 8]
```diff
@@ -627,3 +627,53 @@ class Meta:
...
+    name = graphene.String()
```
Comment (fowczarek) [inline saleor/graphql/translations/types.py:634]: Missing description.

File: saleor/graphql/translations/types.py [hunk 9]
```diff
@@ -627,3 +627,53 @@ class Meta:
...
+    description = JSONString(
+        description="Translated description of the promotion." + RICH_CONTENT
+    )
+
+    class Meta:
```
Comment (fowczarek) [inline saleor/graphql/translations/types.py:639]: Missing description.

File: saleor/graphql/translations/types.py [hunk 10]
```diff
@@ -627,3 +627,53 @@ class Meta:
...
+        model = discount_models.Promotion
+        interfaces = [graphene.relay.Node]
+
+
+class PromotionTranslatableContent(ModelObjectType[discount_models.Promotion]):
+    id = graphene.GlobalID(required=True)
```
Comment (fowczarek) [inline saleor/graphql/translations/types.py:645]: Missing description.

File: saleor/graphql/translations/types.py [hunk 11]
```diff
@@ -627,3 +627,53 @@ class Meta:
...
+    name = graphene.String(required=True)
```
Comment (fowczarek) [inline saleor/graphql/translations/types.py:646]: Missing description.

File: saleor/graphql/translations/types.py (hunk 3 above)
Comment (fowczarek) [inline saleor/graphql/translations/types.py:664]: Missing description.

File: saleor/graphql/translations/types.py [hunk 12]
```diff
@@ -627,3 +627,53 @@ class Meta:
...
+        description="Translated description of the promotion rule." + RICH_CONTENT
+    )
+
+    class Meta:
+        model = discount_models.PromotionRule
+        interfaces = [graphene.relay.Node]
+
+
+class PromotionRuleTranslatableContent(ModelObjectType[discount_models.Promotion]):
+    id = graphene.GlobalID(required=True)
```
Comment (fowczarek) [inline saleor/graphql/translations/types.py:670]: Missing description.

File: saleor/graphql/translations/types.py [hunk 13]
```diff
@@ -627,3 +627,53 @@ class Meta:
...
+    name = graphene.String(required=True)
```
Comment (fowczarek) [inline saleor/graphql/translations/types.py:671]: Missing description.

File: saleor/graphql/translations/types.py (hunk 4 above)
Comment (fowczarek) [inline saleor/graphql/translations/types.py:677]: Missing description.

**Review by fowczarek — COMMENTED** [review]
Missing PR description.

**Review by IKarbowiak — APPROVED** [review]

**Pull request merged**